### DISCUSSION TITLE
show tags in commit history

### DISCUSSION THREAD

**Discussion by SvenStaehs** [community/community#58544] [Repositories]
### Select Topic Area

Product Feedback

## Request
In the Commit History page, I would like to see which commits are tagged, what is the name of the tag (and if it's an annotated tag, show the message when mouse hovers over the tag name)

For example, in this view, I would like to see that the top-most commit is tagged as v1.0.1:
https://github.com/github/docs/commits/a2f178351afc3ad570c5089a9deec80bc9e5f0e7
I can clearly see that tag relation if I open [the commit itself](https://github.com/github/docs/commit/a2f178351afc3ad570c5089a9deec80bc9e5f0e7), but want to see it in the history as well so I can easily see which commits were included in v1.0.0 and which commits are part of v1.0.1

## See also:
- [similar request in refined-git](https://github.com/refined-github/refined-github/issues/807), implemented in 2019 (but I don't want to install tons of browser plugins on my work machine...)
- [same feature is available in Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/repos/git/git-tags?view=azure-devops&tabs=browser#view-tags-in-the-commits-view)

**Comment by rr-it** [comment]
The commit history timeline has dividers by date. How about adding dividers by tag:

![grafik](https://github.com/community/community/assets/29690343/73aa67e4-941b-4129-acec-b1d8f861a596)

**Comment by jason-berk-k1x** [comment]
I'm astonished this isn't just the default?!?!?!  What am I missing???  Why, when you are looking at the commit history, would you *NOT* want to see the tagged commits.  This is one case where Azure DevOps is better than GitHub

**Comment by ofer-pd** [comment]
I had to write my own CLI tool for this because GitHub doesn't provide this very basic, obvious, and essential view on history.

I don't get it.

**Comment by najibghadri** [comment]
This is so funny 🤣 🤣 🤣  The Github Desktop app has this feature but not the web Github commit history 🤣 🤣 🤣  COME ON GUYS! @github

<img width="356" alt="Screenshot 2025-01-15 at 00 04 56" src="https://github.com/user-attachments/assets/e7578b3d-026c-4f80-9369-10f6803314a2" />

**Reply by ofer-pd** [reply]
Thanks for the tip.

**Comment by ImanKCO** [comment]
The fact that even the broken BitBucket supports this and Github doesn't is beyond imagination. That should a relatively easy to add feature.

**Comment by kjpluck** [comment]
Adding my support to this very basic need.

**Comment by jjosenaldo** [comment]
+1, this would be  pretty useful

**Comment by TechnicalFox** [comment]
IMHO this is expected functionality, and it's a shame it's not included in "enterprise" grade software. Personally I don't even need links or annotations, but the commit history page should not be missing default functionality from a standard "git log" call... That is to say; at the very least I expect the plaintext git tag names associated with each ref.

**Comment by rozalily** [comment]
This is also a feature in ADO, already. Though, don't take that as an endorsement of it...

**Comment by clar-gutierrez** [comment]
+1, please add this feature!

**Comment by mgollo** [comment]
We recently migrated hundreds of repos from discontinued Bitbucket server and this is really something I miss badly in GitHub!

**Comment by carlosmaranje** [comment]
Gitlab 1 - 0 Github

**Comment by chrissi-p-bosch** [comment]
Same her. Migrated form BB to GH and realy bad experience not to see tags next to commits

**Comment by smartin-fr** [comment]
Bumping -- will this be a thing?

**Comment by Niketin** [comment]
Please add this

**Comment by kandji-casey** [comment]
please add!

**Comment by SvenStaehs** [comment]
If you wish to show your interest in a feature, please _upvote_ the post itself. AFAIK that's the metric GH team goes by, and writing "+1" comments just spams all the people who want to be notified about news on this topic 😉 Thank you for your interest!

**Comment by Top-Dev1219** [comment]
Adding tag visibility to the Commit History view would be a very useful improvement — especially for developers managing releases or maintaining multiple versions of a project. Tags are often used as lightweight release markers, so surfacing them directly in the commit list would make navigating a repo’s history much easier and more informative.

1. Why It’s Valuable

- Better release context:
Seeing tags like v1.0.0, v1.0.1, etc., next to commits gives immediate context for what code changes belong to each release — without having to open each commit or check the “Tags” page separately.
 
- Improved traceability:
  Teams that follow semantic versioning or release automation can quickly correlate commits with deployment points. This helps when debugging regressions or comparing versions.
 
- Saves clicks:
 Right now, as you said, you can only see tags if you open the individual commit page - which interrupts the flow when browsing through history.

2. Suggested Implementation Details

- Show small tag labels (like colored badges) next to commits that have tags.

- On hover, display: Tag type (lightweight vs annotated), annotated tag message (if available), optionally, creation date and author.

- Could also include a filter (e.g., “Show only tagged commits”) for release review workflows.

3. Developer Experience Impact

- This feature would particularly help: Maintainers doing release audits, CI/CD engineers validating deployment triggers, Open-source contributors understanding project version flow

**Comment by chrisk314** [comment]
I have to switch back and forth between the commits view and the tags view to see which semver tag corresponds to which commit. Very poor UX. What an embarrassment to lack such a basic and obvious feature which is part of the standard `git hist` command output.

**Comment by sarev** [comment]
I'll just bump this one, as another year has passed... Now, where's that tumbleweed emoji?

**Comment by michael-masas-ctrl** [comment]
PLEASE ADD !!! THIS IS STILL RELEVANT

**Comment by nse-immeo** [comment]
Just do it.
<img width="1280" height="720" alt="image" src="https://github.com/user-attachments/assets/4c8a460f-1654-4a69-ac05-6a552bd6e5e9" />

**Comment by benjaminclark3325-svg** [comment]
It would definitely improve clarity to surface tag names and hover messages directly in the commit history view, especially when tracking version groupings like v1.0.1, kind of like how niche needs such as [show dog relocation services india](https://sl1nk.com/1z7n99m) benefit from having all key details visible at a glance.

**Comment by dbpieter** [comment]
Missing this feature from ADO.
Guess I'll have to do it the old school way:  git log --all --decorate --graph --oneline

**Comment by yoshida-yuen-ijs** [comment]
Gitea can do this.
I was surprised that Github couldn't do this.

**Comment by Kludgy4** [comment]
Yeah please add this

**Comment by Permillisecond** [comment]
After migrating from ADO, it's really painful to look at history without tags. Github pls do this 🙏

**Comment by wrparker** [comment]
+1 please do this otherwise i'm going to have to write a browser extension and/or switch to gitlab
